src/Service/Kraken/KrakenBuyService.php

```diff
@@ -16,34 +16,54 @@ class KrakenBuyService implements BuyServiceInterface
     protected string $lastUserRef;
     protected KrakenClientInterface $client;
     protected string $baseCurrency;
-    protected string $tradingPair;
 
     public function __construct(KrakenClientInterface $client, string $baseCurrency)
     {
         $this->client = $client;
         $this->baseCurrency = $baseCurrency;
-        $this->tradingPair = sprintf('XBT%s', $this->baseCurrency);
     }
 
     public function supportsExchange(string $exchange): bool
     {
         return 'kraken' === $exchange;
     }
 
-    public function initiateBuy(int $amount): CompletedBuyOrder
+  /**
+   * @param int $amount
+   * @param bool $simulate
+   *   Set this boolean to true to simulate an buy order - see https://support.kraken.com/hc/en-us/articles/360000919926-Does-Kraken-offer-an-API-test-environment-
+   *
+   * @return CompletedBuyOrder
+   * @throws PendingBuyOrderException
+   */
+    public function initiateBuy(int $amount, string $asset, bool $simulate = false): CompletedBuyOrder
```

Comment:
Je wijkt hier af van de interface wat in de regel niet zo'n goed idee is, buiten development is er eigenlijk ook geen reden om de aankoop te simuleren.

Reply:
De `$simulate` moet er uit inderdaad. Icm Docker die ziet of je in de development stage zit, zouden we dit wel kunnen gebruiken?

Reply:
Als je er op staat om perse een simulate optie er in te houden zou ik een extra service introduceren, deze kan dan de KrakenBuyService extenden of decoraten. 

Door middel van prioriteiten aan je tags mee te geven kun je dan je `KrakenSimulatedBuyService` net boven de andere `KrakenBuyService` zetten. In de `supports` methode kun je dan controleren op de aanwezigheid van bijvoorbeeld `KRAKEN_SIMULATE_BUY` environment variabele. 

Dit zou dan ongeveer zo uitwerken:

`$ KRAKEN_SIMULATE_BUY=1 bin/bitcoin-dca buy 10 --yes`